### PR TITLE
Simplify how environment variables are used

### PR DESCRIPTION
**Pull request by jkenny**
Rather than creating .list files and pushing the contents into variables in the start_docker_container script, which requires having to do evals, just create one script to set all these variables and source it in the start_docker_container script

### CODE REVIEW

File: nova/core/deploy/deploy_stack.py [hunk 1]
```diff
@@ -162,49 +162,41 @@ def __create_app_spec(self):
...
+        env_vars_script += 'DOCKER_VARS="%s"\n' % deployment_variables.strip()
 
         deployment_arguments = ''
         if stack.deployment_arguments is not None:
             for opts in stack.deployment_arguments:
                 for args in opts.items():
                     deployment_arguments += "{0}={1} ".format(args[0], args[1])
-            deployment_arguments += "\n"
-        self.__render_file('env-vars/%s/docker-args.list' % stack.stack_type, deployment_arguments.strip())
+        env_vars_script += 'DOCKER_ARGS="%s"\n' % deployment_arguments.strip()
```
Comment (andreyk0) [inline nova/core/deploy/deploy_stack.py]: Sorry, maybe I'm not reading it correctly but isn't this just going to generate a bunch of lines that set same variable? If so - only the last line's value will be set.

You probably want the result to be something like:

```bash
DOCKER_ARGS="$DOCKER_ARGS -arg1 val1"
DOCKER_ARGS="$DOCKER_ARGS -arg2 val2"

```

**Comment by andreyk0** [comment]
👍 nice, one comment inline

File: nova/core/deploy/deploy_stack.py (hunk 1 above)
Comment (jkenny) [inline nova/core/deploy/deploy_stack.py]: I *think* I'm adding a new line each time to the env_vars_script variable. (using +=) Each line sets a different variable.

Comment (andreyk0) [inline nova/core/deploy/deploy_stack.py]: K, all good then :) I guess just try it with a multi-arg case and see what you get.

Comment (jkenny) [inline nova/core/deploy/deploy_stack.py]: Multiple args should get concatenated on line 193 and then they're assigned to the DOCKER_ARGS variable in one go in env_vars_script on line 194. So DOCKER_ARGS should only appear once.
Not sure was that what you were asking :) Anyway, I'll test it out...

**Comment by andreyk0** [comment]
👍 LGTM

**Pull request merged**